### PR TITLE
convert session cacher into using locks

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco not a priority, but this converts `SessionCacher` to using locks instead of the channel loop, it's just much clearer this way.

### CODE REVIEW

**Comment by maxtaco** [comment]
I'm on board, but @patrickxb can you give this a second set of eyes?

Also, does it pass `go vet`?

File: rpc/server/session_cacher.go [hunk 1]
```diff
@@ -138,78 +125,56 @@ func (sc *SessionCacher) clearExpiryQueue() {
...
 // AuthenticateSessionToken authenticates a given session token, first against
 // the cache and the, if that fails, against the parent AuthInterface.
 func (sc *SessionCacher) AuthenticateSessionToken(ctx context.Context, tok gregor1.SessionToken) (res gregor1.AuthResult, err error) {
 
-	respCh := make(chan *gregor1.AuthResult)
-
 	// Fire off request for a cached session
 	sc.stats.Count("AuthenticateSessionToken")
-	sc.reqCh <- readTokReq{tok, respCh}
+	sc.RLock()
```
Comment (maxtaco) [inline rpc/server/session_cacher.go]: you can wind up with multiple guys working on `tok` at the same time here.  I guess that's not the end of the world, and the old code had this problem too?

**Comment by mmaxim** [comment]
`go vet` seems cool with it.

File: rpc/server/session_cacher.go [hunk 2]
```diff
@@ -112,21 +98,22 @@ func (sc *SessionCacher) deleteSID(sid gregor1.SessionID) {
...
 }
 
 // clearExpiryQueue removes all currently expired sessions.
 func (sc *SessionCacher) clearExpiryQueue() {
+	sc.Lock()
+	defer sc.Unlock()
+
 	now := sc.cl.Now()
 	for _, e := range sc.expiryQueue {
 		if now.Before(e.expiry) {
```
Comment (patrickxb) [inline rpc/server/session_cacher.go]: I know this isn't part of this PR, but `sc.deleteSID(e.sid)` seems clearer to me...

File: rpc/server/session_cacher.go [hunk 3]
```diff
@@ -79,30 +76,19 @@ func (sc *SessionCacher) updateStats() {
...
 	sc.sessionIDs[res.Sid] = tok
 }
 
-type deleteSIDReq struct {
-	sid gregor1.SessionID
-}
-
 func (sc *SessionCacher) deleteSID(sid gregor1.SessionID) {
 	if tok, ok := sc.sessionIDs[sid]; ok {
 		if _, ok := sc.sessions[tok]; ok {
```
Comment (patrickxb) [inline rpc/server/session_cacher.go]: again, not in this PR, but just `delete(sc.sessions, tok)` is fine.  no need to check if it is already there.

**Comment by patrickxb** [comment]
LGTM!  Much easier to follow.

**Pull request merged**